File: Scripts/volatile_and_force_compiler_task/inheritance-single/java/InheritanceSingle.java
import java.util.*;

class Animal{
   //functions go here...
   public void speak(){
       System.out.println("Animal speaking");
   }

   public void walk(){
       System.out.println("Animal walking");
   }
}

 class Collie extends Dog{
    //functions go here...
    public void speak(){
        System.out.println("Collie barking");
    }
 }
 
  class Lab extends Dog{
    //functions go here...
    public void speak(){
        System.out.println("Lab barking");
    }
 }
 
  class Cat extends Animal{
    //functions go here...
    public void speak(){
        System.out.println("Cat meawing");
    }
 }

  class Dog extends Animal{
    //functions go here...
    public void speak(){
        System.out.println("Dog barking");
    }
 }

 public class InheritanceSingle {


    private static volatile Dog a = new Dog();

    public static void main(String[] Args) {

        for (int i=0; i<1000000; ++i) {
            a.speak();
            a.walk();
        }
    }
 }
 

File: scripts/volatile_and_force_compiler_task/inheritance-single/c++/inheritance-single.cpp
#include <iostream>
using namespace std;

class Animal
{
  // ... 
  public:
     void speak(){
        cout<<"Animal speaking\n";
    }
    void walk(){
        cout<<"Animal walking\n";
    }
};
 
class Dog: public Animal
{
  // ... 
public:
  void speak(){
    cout<<"Dog barking\n";
}
};
 
class Lab: public Dog
{
  // ...
public:
  void speak(){
    cout<<"Lab barking\n";
}
};
 
class Collie: public Dog
{
  // ...
public:
  void speak(){
    cout<<"Collie barking\n";
}
};
 
class Cat: public Animal
{
  // ...
public:
  void speak(){
    cout<<"Cat meawing\n";
}
};

int main () {
    for (int i = 0; i < 1000000; ++i) {
        Dog a;
        a.speak();
        a.walk();
    }
    return 0;
}
